### PR TITLE
Reserve locale and htmlSafe attributes

### PR DESCRIPTION
**Pull request by jasonmit**
Based on our discussion over slack /cc @jamesarosen 

This is for handling deprecations ahead of 5.x

Loosely related to #377

### CODE REVIEW

File: tests/unit/i18n-t-test.js [hunk 1]
```diff
@@ -36,6 +40,30 @@ test('returns "missing translation" translations', function(assert) {
...
+    }
+  };
+
+  service.t('not.yet.translated', { htmlSafe: true });
+  assert.equal(deprecations, 1);
+
+  service.t('not.yet.translated', { locale: true });
+  assert.equal(deprecations, 2);
+
+  service.t('not.yet.translated');
```
Comment (jamesarosen) [inline tests/unit/i18n-t-test.js]: Maybe `service.t('not.yet.translated', { some: 'other key' })`

Comment (jasonmit) [inline tests/unit/i18n-t-test.js]: Sounds good, I'll add one that includes interpolation

Comment (jasonmit) [inline tests/unit/i18n-t-test.js]: All done, it's on line 63 to catch both w/ and w/o data attribute.

**Comment by jamesarosen** [comment]
+1

**Comment by jasonmit** [comment]
@jamesarosen once merged I'll start on the second part of what we discussed, supporting inlining a locale when looking up a translation.

**Pull request merged**